### DISCUSSION TITLE
Linux: Use SDR colorspace or enable controlling brightness of HDR brightness

### DISCUSSION THREAD

**Discussion by Cybolic** [ghostty-org/ghostty#3932] [Feature Requests, Ideas] [answered]
As described [on Discord](https://discord.com/channels/1005603569187160125/1322696060170408079), Ghostty currently opens in a high bit depth / HDR window at full brightness when running on an HDR enabled monitor, making it searingly bright to look at and quite a bit oversaturated (at least on NixOS / Plasma 6).

I assume the completion of #2125 might eventually fix this, but in the meantime it would be very nice to be able to launch Ghostty in a lower bit depth window using a flag such as `--use-sdr`.

Currently, I'm working around this by using a tone mapping shader, but it's not perfect and Ghostty's native elements (window bar, tabs, messages, etc.) are still blindingly bright, so it's not a proper workaround.

[Log attached for good measure.](https://github.com/user-attachments/files/18270658/ghostty.log)

**Comment by Cybolic** [comment, marked as answer]
After some investigation, this seems to be an issue [with all Gtk4 apps on HDR displays](https://bugs.kde.org/show_bug.cgi?id=495647). The suggested workaround of `GDK_DISABLE=color-mgmt` did not work for me, but the more aggressive `GDK_BACKEND=x11` did.
Ghostty is not to blame; let's hope Gtk gets this sorted out sooner rather than later.

I'll leave this open since the issue still exists, even if it's not in Ghostty's codebase.

**TL;DR**: Run with `GDK_BACKEND=x11 ghostty`

**Comment by 13k** [comment]
HDR breaks not only brightness, but also font rendering and colors.

All images are using `theme = "tokyonight"`, KDE Plasma 6.2.5 on Wayland.

* HDR=on; `gtk-gsk-renderer = "opengl"`:

![Screenshot_20250205_102109](https://github.com/user-attachments/assets/9f5fcc5e-4628-426f-bed4-80a2f9c52170)

* HDR=on; `gtk-gsk-renderer = "default"`; `GSK_DEBUG=renderer`:

```console
Not using Vulkan: Vulkan support disabled via GDK_DISABLE
Using renderer 'GskNglRenderer' for surface 'GdkWaylandToplevel'
```

![Screenshot_20250205_102829](https://github.com/user-attachments/assets/89f826d6-87ac-44e2-8547-ed831d860633)

* HDR=**off**; `gtk-gsk-renderer = "opengl"`:

![Screenshot_20250205_102520](https://github.com/user-attachments/assets/73ea8da9-50c0-4304-b8b9-ffb24acf19c9)

nvim (also `colorscheme=tokyonight`) looks even more egregious.

* kitty:

![Screenshot_20250205_104148](https://github.com/user-attachments/assets/0cf46b06-63eb-4fef-9c50-5f2b83cdb296)

* ghostty:

![Screenshot_20250205_104151](https://github.com/user-attachments/assets/a8596642-4d75-4598-8784-6e539ff65ced)

Using {HDR=on, `gtk-gsk-renderer = "default"`} renders (almost?) correctly, but falls into the [ngl/vulkan renderer issue losing context when closing tabs](https://github.com/ghostty-org/ghostty/issues/1667), so it's unusable. The only viable option is to turn off HDR.

It seems the new Gtk renderers (ngl/vulkan) can handle HDR and other modern features, such as fractional scaling and better antialias, but ghostty is stuck using the old `gl` renderer for reasons I don't actually understand.

**Comment by oekelly** [comment]
Hi All,

I've just encountered this issue also. I just updated the latest version of Nobara (FC41 based: 6.13.2-200.nobara.fc41.x86_64), which uses Wayland with KDE Plasma 6.3.0.

Upon updating and opening Ghostty (version 1.1.0-1.fc41):

- the titlebar decorations (New tab, View open tabs, Main menu) are removed, and the the system default titlesbar is used instead;
- the colour scheme was completely washed out. I was using the Darkside scheme for a nice dark grey, and instead it appears almost cloud grey - very light.

Setting `export GDK_BACKEND=x11` and rebooting resolved both of the above issues. 

As far as am I aware I do not have a HDR compatible monitor, i'm using 2x Dell 27" s2719DGF (https://www.displayspecifications.com/en/model/76521490)

**Reply by jcollie** [reply]
This is a bug in GTK when it's running under KDE 6.3. This should be fixed in 1.1.1 or 1.1.2 (released earlier today).
